docs/proposals/maintenance.md

```diff
@@ -0,0 +1,179 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest release of this document can be found
+[here](http://releases.k8s.io/release-1.1/docs/proposals/maintenance.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Node Maintenance
+
+## Abstract
+
+A proposal for implementing safe node maintenance primitives, in stages.
+
+## Overview
+
+Nodes need periodic maintenance, most of which require partial or total
+unavailability to pods.  The cluster admin would like for this to happen (1)
+automatically, and (2) respecting any service quality constraints.
+
+Examples of maintenance:
+
+- unmount/remount filesystems to change mkfs params
+- remove from service but leave up for debugging
+- reboot for kernel or image upgrade
+- remove from service for hardware repair
+- remove from service permanently
+
+## Service Quality
+
+This may be thought of as the cell admin offering a guarantee or SLO, or as the
```

Comment:
On Wed, Nov 18, 2015 at 10:18 PM, Paul Morie <notifications@github.com>
wrote:

> In docs/proposals/maintenance.md
> <https://github.com/kubernetes/kubernetes/pull/17393#discussion_r45305996>
> :
>
> > +
> > +Nodes need periodic maintenance, most of which require partial or total
> > +unavailability to pods.  The cluster admin would like for this to happen (1)
> > +automatically, and (2) respecting any service quality constraints.
> > +
> > +Examples of maintenance:
> > +
> > +- unmount/remount filesystems to change mkfs params
> > +- remove from service but leave up for debugging
> > +- reboot for kernel or image upgrade
> > +- remove from service for hardware repair
> > +- remove from service permanently
> > +
> > +## Service Quality
> > +
> > +This may be thought of as the cell admin offering a guarantee or SLO, or as the
>
> s/cell/node, or am i missing something?
>

Fixed to 'cluster admin'.  Borg terminology was seeping in.


> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/17393/files#r45305996>.
>